pkg/server/ca/manager.go

```diff
@@ -811,6 +818,40 @@ func (m *Manager) notify(ctx context.Context, event string, advise bool, pre fun
 	return nil
 }
 
+func (m *Manager) filterInvalidEntries(ctx context.Context, entries *journal.Entries) ([]*JWTKeyEntry, []*X509CAEntry, error) {
+	bundle, err := m.fetchOptionalBundle(ctx)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	filteredEntriesJwtKeys := []*JWTKeyEntry{}
+	filteredEntriesX509CAs := []*X509CAEntry{}
+
+	// certificate append to the database whenever upstreamClient does not exist
+	if m.upstreamClient != nil {
+		filteredEntriesX509CAs = entries.X509CAs
+	} else {
```

Comment:
If we move the x509 logic below the jwt logic, we can avoid this if/else and improve readability

```go
for _, entry := range entries.GetJwtKeys() {
     ...
}

// If we have an upstream authority then we're not recovering a root CA, so we do
// not expect to find our CA certificate in the bundle. Simply proceed.
if m.upstreamClient != nil {
     return filteredEntriesJwtKeys, entries.X509CAs, nil
}

for _, entry := range entries.GetX509CAs() {
     ...
}
```

Reply:
Make sense, that I have finished these code structure suggestions and made changes.